File: GFG/Python/2025-08-05_palindrome_sentence.py
# GFG - Palindrome Sentence
# Date: 2025-08-05

class Solution:
    def isPalinSent(self, s):
        # code here
        n = len(s)
        i = 0
        j = -1

        while i - n < j:
            while i < n and not s[i].isalnum():
                i += 1
            while abs(j) <= n and not s[j].isalnum():
                j -= 1
            if i - n >= j:
                break
            if s[i].lower() != s[j].lower():
                return False
            i += 1
            j -= 1

        return True

File: GFG/C++/2025-08-05_palindrome_sentence.cpp
// GFG - Palindrome Sentence
// Date: 2025-08-05

class Solution {
  public:
    bool isPalinSent(string &s) {
        // code here
        int n = s.length();
        int i = 0;
        int j = n - 1;

        while (i <= j) {
            while (i < j && !isalnum(s[i])) i++;
            while (i < j && !isalnum(s[j])) j--;

            if (tolower(s[i]) != tolower(s[j])) return false;

            i++;
            j--;
        }

        return true;
    }
};
